blaze/server/spider.py

```diff
@@ -105,9 +104,27 @@ def from_yaml(path, ignore=(ValueError, NotImplementedError), followlinks=True,
     data_spider : Traverse a directory tree for resources
     """
     resources = {}
+    yaml_dir = os.path.split(os.path.abspath(path.name))[0]
+    owd = os.getcwd()
     for name, info in yaml.load(path.read()).items():
         try:
             source = info.pop('source')
+
+            # convert relative pathes to absolute relative to yaml file
+            if (not os.path.isabs(source) and
+               not source.startswith('http') and
+               '~' not in source):
+                os.chdir(yaml_dir)
+                src = os.path.abspath(source)
+                os.chdir(owd)
+
+                # handle 'sqlite:///../path/with/protocol/test.db'
+                if ('://' in source):
```

Comment:
Cases like this make me think we should just bracket this block with a chdir. This way we don't need to manipulate the paths directly.